src/org/mozilla/fxtest/ServiceBook.groovy

```diff
@@ -0,0 +1,168 @@
+    HttpResponse resp = new HttpResponse(connection);
+
+    if(resp.isFailure()){
+        error("\nGET from URL: $requestUrl\n  HTTP Status: $resp.statusCode\n  Message: $resp.message\n  Response Body: $resp.body");
+    }
+    return resp;
+}
+
+HttpResponse doPostHttpRequestWithJson(String json, String requestUrl){
+    return doHttpRequestWithJson(json, requestUrl, "POST");
+}
+
+HttpResponse doPutHttpRequestWithJson(String json, String requestUrl){
+    return doHttpRequestWithJson(json, requestUrl, "PUT");
+}
+
+HttpResponse doHttpRequestWithJson(String json, String requestUrl, String verb){
+    URL url = new URL(requestUrl);
+    HttpURLConnection connection = url.openConnection();
+
+    connection.setRequestMethod(verb);
+    connection.setRequestProperty("Content-Type", "application/json");
+    connection.doOutput = true;
+
+    def writer = new OutputStreamWriter(connection.outputStream);
+    writer.write(json);
+    writer.flush();
+    writer.close();
+
+    connection.connect();
+
+    //parse the response
+    HttpResponse resp = new HttpResponse(connection);
+
+    if(resp.isFailure()){
+        error("\n$verb to URL: $requestUrl\n    JSON: $json\n    HTTP Status: $resp.statusCode\n    Message: $resp.message\n    Response Body: $resp.body");
+    }
+
+    return resp;
+}
+
+class HttpResponse {
+    String body;
+    String message;
+    Integer statusCode;
+    boolean failure = false;
+
+    public HttpResponse(HttpURLConnection connection){
+        this.statusCode = connection.responseCode;
+        this.message = connection.responseMessage;
+
+        if(statusCode == 200 || statusCode == 201){
+            this.body = connection.content.text;//this would fail the pipeline if there was a 400
+        }else{
+            this.failure = true;
+            this.body = connection.getErrorStream().text;
+        }
+
+        connection = null; //set connection to null for good measure, since we are done with it
+    }
+}
+
+
+// Servicebook iterator to get operational tests for a given project
+Object getProjectTests(String name) {
+    resp = doGetHttpRequest("http://servicebook.dev.mozaws.net/api/project").body;
+    def jsonSlurper = new JsonSlurperClassic();
+    def projects = jsonSlurper.parseText(resp);
+
+    for (project in projects.data) {
+        def projectName = name.toLowerCase();
+        if (project.name == projectName) {
+            echo "The project " + name + " was found!"
+
+            def jenkin_tests = [];
+                for (test in project.tests) {
+                    if (test.jenkins_pipeline) {
+                        echo "Adding one test " + test.name
+                        jenkin_tests << test;
+                    }
+                }
+            return jenkin_tests;
+        }
+    }
+    echo "The project " + name + " was not found"
+    return null;
+}
+
+
+
+def validURL(url) {
+    allowedOrgs = ['Kinto', 'mozilla', 'mozilla-services', 'tarekziade', 'rpappalax']
```

Comment:
This restriction is good to be aware of. I was thinking folks could make sure tests were somewhat viable before pushing them to a mozilla* org (i.e. that tests could be triggered from any org), but this is much safer route.   I suppose we can still follow that workflow in PreProdJenkins and do soak testing there (without hooking into pipeline).
If you don't need the org for tarekziade or me, I'd go ahead and remove them.

Reply:
Thanks, Richard.  Actually, yes: now that we're getting really close to using this in a production/production-like environment and workflow, it's probably a good time to reel the validation back a bit; I'll make that change now.